### PR TITLE
ENH: Using built-in round on a series #11763

### LINKED ISSUE #11763: ENH: Using built-in round on a series
Is there some philosophical reason that round(Series) and round(Dataframe) is not defined?  See the example below.  Being able to write round() as a function of a Dataframe or Series, as opposed to doing it afterwards would be nice.  I'm willing to implement `__round__()` to make it work, but don't want to go down that path if there is some reason `__round__()` is not implemented.

Example:
```
    import pandas as pd
    import numpy as np
    df =  pd.DataFrame(np.random.randn(10, 3), columns=['A', 'B', 'C'])
    print (df.A.round())
    print (round(df))
```
Produces
```
0   -2
1    1
2   -1
3   -1
4    2
5    0
6    0
7    0
8   -2
9    0
Name: A, dtype: float64
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-9d2be9cbdc70> in <module>()
      3 df =  pd.DataFrame(np.random.randn(10, 3), columns=['A', 'B', 'C'])
      4 print (df.A.round())
----> 5 print (round(df))

TypeError: type DataFrame doesn't define __round__ method
```

### PR DESCRIPTION
**Pull request by Dr-Irv**
closes #11763 

This contains changes to support `round()` (works in Python 3 only) for DataFrame, Series and Panel. Also includes `Panel.round()`.  Note on code changes - moved code from test_format.py to test_frame.py because it made more logical sense to put the testing code for `round()` into test_frame.py

### CODE REVIEW

File: doc/source/whatsnew/v0.18.0.txt [hunk 1]
```diff
@@ -81,6 +82,17 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
    s = Series(dr)
    s
    s.dt.round('D')
+   
+Built-in round() support
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: just a single line in the enhancements section is enough

File: pandas/tests/test_frame.py [hunk 2]
```diff
@@ -13391,6 +13391,235 @@ def wrapper(x):
...
+
+        dfs = pd.concat((df, df), axis=1)
+        rounded = dfs.round()
+        self.assertTrue(rounded.index.equals(dfs.index))
+
+        decimals = pd.Series([1, 0, 2], index=['A', 'B', 'A'])
+        self.assertRaises(ValueError, df.round, decimals)
+        
+    def test_built_in_round(self):
+        if sys.version < LooseVersion('3.0'):
```
Comment (jreback) [inline pandas/tests/test_frame.py]: use ``compat.PY3``

File: pandas/tests/test_frame.py [hunk 3]
```diff
@@ -13391,6 +13391,235 @@ def wrapper(x):
...
+            raise nose.SkipTest('build in round cannot be overriden prior to Python 3')
+
+        # GH11763
```
Comment (jreback) [inline pandas/tests/test_frame.py]: to be honest you don't need any of these tests, they are all covered above, or is there something else that you are trying to do?

File: pandas/tests/test_series.py [hunk 4]
```diff
@@ -3008,6 +3008,27 @@ def test_round(self):
...
         self.assertEqual(result.name, self.ts.name)
+ 
+        s = Series([1.123, 2.123, 3.123], index=lrange(3))               
+        outarr = np.empty(3)
+        result = s.round(out=outarr)
+        # outarr should be modified
+        self.assert_numpy_array_equal(result, outarr)
+        
+    def test_built_in_round(self):
+        if sys.version < LooseVersion('3.0'):
```
Comment (jreback) [inline pandas/tests/test_series.py]: use ``compat.PY3`` here

File: pandas/tests/test_series.py [hunk 5]
```diff
@@ -3008,6 +3008,27 @@ def test_round(self):
                           name='ts')
         assert_series_equal(result, expected)
         self.assertEqual(result.name, self.ts.name)
+ 
+        s = Series([1.123, 2.123, 3.123], index=lrange(3))               
+        outarr = np.empty(3)
+        result = s.round(out=outarr)
```
Comment (jreback) [inline pandas/tests/test_series.py]: so drop the ``out`` paramater from ``Series.round`` (you can put that in API changes). We don't support this ANYWHERE else so its really inconsitent.

File: pandas/tests/test_series.py [hunk 6]
```diff
@@ -3008,6 +3008,27 @@ def test_round(self):
...
+        # outarr should be modified
+        self.assert_numpy_array_equal(result, outarr)
+        
+    def test_built_in_round(self):
+        if sys.version < LooseVersion('3.0'):
+            raise nose.SkipTest('build in round cannot be overriden prior to Python 3')
+        
+        s = Series([1.123, 2.123, 3.123], index=lrange(3))
+        result = round(s)
```
Comment (jreback) [inline pandas/tests/test_series.py]: futher there is already a ``test_round`` test, at the very least put these right after it

Comment (Dr-Irv) [inline pandas/tests/test_series.py]: The code above is the end of ``test_round`` and the beginning of ``test_built_in_round``

Comment (jreback) [inline pandas/tests/test_series.py]: ahh ok

File: pandas/core/frame.py [hunk 7]
```diff
@@ -4392,8 +4396,8 @@ def _dict_round(df, decimals):
                     raise ValueError("Index of decimals must be unique")
             new_cols = [col for col in _dict_round(self, decimals)]
         elif com.is_integer(decimals):
-            # Dispatch to numpy.round
-            new_cols = [np.round(v, decimals) for _, v in self.iteritems()]
+            # Dispatch to Series.round
+            new_cols = [v.round(decimals) for _, v in self.iteritems()]
```
Comment (jreback) [inline pandas/core/frame.py]: this is fine, but not really necessary as these do the same thing

File: doc/source/whatsnew/v0.18.0.txt [hunk 8]
```diff
@@ -33,6 +33,8 @@ Other enhancements
 
 - Handle truncated floats in SAS xport files (:issue:`11713`)
 - ``read_excel`` now supports s3 urls of the format ``s3://bucketname/filename`` (:issue:`11447`)
+- A simple version of Panel.round() is now implemented (:issue:`11763`)
+- For Python 3.x, round(DataFrame), round(Series), round(Panel) will work (:issue:`11763`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: in API breaking show that we removed the ``out`` arg from ``Series.round`` (same issue number)

use double backticks no things like ``round(DataFrame)`` and such

**Comment by jreback** [comment]
small docs changes. pls squash. ping when green.

File: pandas/core/frame.py (hunk 7 above)
Comment (Dr-Irv) [inline pandas/core/frame.py]: Actually, this change was needed to get rid of the ``out`` argument. If ``DataFrame.round()`` calls ``np.round()``, then ``np.round()`` dispatches to ``Series.round()``, passing the ``out`` argument.

Comment (jreback) [inline pandas/core/frame.py]: ok

**Comment by Dr-Irv** [comment]
@jreback It's squashed now.  Not sure what to do next to force the Travis CI thing to happen since it says there are conflicts

**Comment by jreback** [comment]
you need to rebase on master as well

**Comment by Dr-Irv** [comment]
@jreback I think this is good to go.

File: pandas/tests/test_panel.py [hunk 9]
```diff
@@ -444,7 +444,7 @@ def test_abs(self):
         assert_series_equal(result2, expected)
         self.assertEqual(result.name, 'A')
         self.assertEqual(result2.name, 'A')
-
+
```
Comment (jreback) [inline pandas/tests/test_panel.py]: delete this line

File: pandas/core/panel.py [hunk 10]
```diff
@@ -622,6 +622,34 @@ def head(self, n=5):
...
+        Panel object
+        
+        See Also
+        --------
+        numpy.around
+        """
+        if com.is_integer(decimals):
+            result = np.apply_along_axis(np.round, 0, self.values)
+            return self._wrap_result(result, axis=0)
+        else:
```
Comment (jreback) [inline pandas/core/panel.py]: no need for an else, simply raise at the end

File: doc/source/whatsnew/v0.18.0.txt [hunk 11]
```diff
@@ -90,6 +92,8 @@ In addition, ``.round()`` will be available thru the ``.dt`` accessor of ``Serie
 Backwards incompatible API changes
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+The parameter ``out`` has been removed from the ``Series.round()`` method. (:issue:`11763`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.0.txt]: add a '-'

**Comment by jreback** [comment]
couple of trivial changes. ping when pushed (as its already green)

**Comment by Dr-Irv** [comment]
@jreback pushed those changes (and squashed them as well)

**Comment by Dr-Irv** [comment]
@jreback last build failed due to a URL issue.  Not sure how to restart the run

**Comment by jreback** [comment]
@Dr-Irv no worries. thanks for this!

yeh travis fails once in a while on the build step.

**Pull request merged**